k8s/utils/health/cluster_health_check.py

```diff
@@ -0,0 +1,88 @@
+from kubernetes import client, config
+import time
+import argparse
+import signal
+import sys
+
+def timeout_handler(x,y):
+    raise Exception('timeout')
+
+def create_api():
+    while True:
+        try:
+            v1=client.CoreV1Api()
+            break
+        except Exception as e:
+            print "error occurred while creating core v1 api:", e
+            if e == 'timeout':
+                sys.exit(1)                
+            time.sleep(5)
+    return v1
+
+def get_nodes(node_count):
+    v1 = create_api()
+    while True:
+        try:
+            getNodes = v1.list_node()
+            if len(getNodes.items) == int(node_count) + 1:
+                return getNodes.items
+        except Exception as e:
+            print "error occured while getting nodes:", e
+            if e == 'timeout':
+                sys.exit(1)
+            time.sleep(5)
+
+def get_node_status(node_count):
+    count = 0
+    nodes = get_nodes(node_count)
+    for node in nodes:
+        obj = node.status.conditions
+        for i in obj:
+            if i.type == 'Ready':
+                count = count + 1
+    return count
+
+def checkCluster(node_count):
+    while True:
+        try:
+            count = get_node_status(node_count)
+            if count == int(node_count) + 1: # +1 master node
```

Comment:
Will this check for for clusters in GKE as well as baremetal? What if there are 3 master nodes instead of 1?

Reply:
@kmova, in that case, the only thing I can think of, is passing the node counts as a argument